packages/loader/src/types.ts

```diff
@@ -1,6 +1,6 @@
 import { Page, RequestCtx } from "@ekino/rendr-core";
 
-export type Loader = (ctx: RequestCtx) => Promise<Page | void> | void;
+export type Loader<T> = (ctx: RequestCtx) => Promise<T> | void;
```

Comment:
Not sure this this to be a generic, the loader need to return a promise with a Page, or with nothing, if there is no page, it's mean the loader uses the response object from the context to stream a response. Like https://github.com/ekino/rendr/blob/master/examples/nextjs-with-remoteapi/services/api.js#L113-L130

If the page cannot be found in the loader, the loader should throw a NotFoundError. Like in the PR: https://github.com/ekino/rendr/pull/4/commits/e5877de41fad646ac08ffda94e08f803d27590f6#diff-d972cb8a4083394c759e4144ad5d42cbR50

Reply:
I'm using Node version 10.14.2.

Regarding `Loader` generic: this is a consequence of me changing the return type of `pipe` from `Promise<void>` to `Promise<string>`. 

`pipe` is returned from `createApiLoader` under certain conditions, so they need compatible return types. See: https://github.com/ekino/rendr/blob/master/packages/loader/src/services/api.ts#L72-L75. 

In this MR, `pipe` is now trying to do two things. I could separate `pipe` into two different functions for the different uses:

```ts
function pipe(source: Readable): Promise<string> {
  return new Promise<string>((resolve, reject) => {
    let data: string = "";
    source.on("data", (chunk: any) => {
      data += chunk;
    });

    source.on("end", (err: Error) => {
      if (err) {
        reject(err);
      } else {
        resolve(data);
      }
    });
  });
}

function pipeToWritable(source: Readable, dest?: Writable): Promise<void> {
  return new Promise<void>((resolve, reject) => {
    source.pipe(dest);
 
    source.on("end", (err: Error) => {
      if (err) {
        reject(err);
      } else {
        resolve();
      }
    });
  });
}
```

What do you think?